docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,373 @@
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: introduce a warning on the `enabled` and `compatibility` modes of the `-Xjvm-default` compiler option
+> - \>= 1.8.0: raise this warning to an error
+
+### Prohibit calls to functions named `suspend` with a trailing lambda
+
+> **Issue**: [KT-22562](https://youtrack.jetbrains.com/issue/KT-22562)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6 no longer allows calling user functions named `suspend` that have the single argument of a functional type passed as a trailing lambda
+>
+> **Deprecation cycle**:
+>
+> - 1.3.0: introduce a warning on such function calls
+> - 1.6.0: raise this warning to an error
+> - 1.7.0: introduce changes to the language grammar, so that `suspend` before `{` is parsed as a keyword
+
+
+## Standard library
+
+### Gradually change the return type of collection `min` and `max` functions to non-nullable
+
+> **Issue**: [KT-38854](https://youtrack.jetbrains.com/issue/KT-38854)
+>
+> **Component**: kotlin-stdlib
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: the return type of collection `min` and `max` functions will be changed to non-nullable in Kotlin 1.7
+>
+> **Deprecation cycle**:
+>
+> - 1.4.0: introduce `...OrNull` functions as synonyms and deprecate the affected API (see details in the issue)
+> - 1.5.0: raise the deprecation level of the affected API to an error
+> - 1.6.0: hide the deprecated functions from the public API
+> - 1.7.0: reintroduce the affected API but with non-nullable return type
+
+### Deprecate floating-point array functions: `contains`, `indexOf`, `lastIndexOf`
+
+> **Issue**: [KT-28753](https://youtrack.jetbrains.com/issue/KT-28753)
+>
+> **Component**: kotlin-stdlib
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin deprecates floating-point array functions `contains`, `indexOf`, `lastIndexOf` that compare values using the IEEE-754 order instead of the total order
+>
+> **Deprecation cycle**:
+>
+> - 1.4.0: deprecate the affected functions with a warning
+> - 1.6.0: raise the deprecation level to an error
+> - 1.7.0: hide the deprecated functions from the public API
+
+### Migrate declarations from `kotlin.dom` and `kotlin.browser` packages to `kotlinx.*`
+
+> **Issue**: [KT-39330](https://youtrack.jetbrains.com/issue/KT-39330)
+>
+> **Component**: kotlin-stdlib (JS)
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: declarations from the `kotlin.dom` and `kotlin.browser` packages are moved to the corresponding `kotlinx.*` packages to prepare for extracting them from stdlib
+>
+> **Deprecation cycle**:
+>
+> - 1.4.0: introduce the replacement API in `kotlinx.dom` and `kotlinx.browser` packages
+> - 1.4.0: deprecate the API in `kotlin.dom` and `kotlin.browser` packages and propose the new API above as a replacement
+> - 1.6.0: raise the deprecation level to an error
+> - \>= 1.8: remove the deprecated functions from stdlib
```

Comment:
```suggestion
> - \>= 1.8.0: remove the deprecated functions from stdlib; move the API in kotlinx.* packages to a separate library
```

Reply:
I believer it should be fine to use >= 1.8 to indicate less certainty in the version that in >= 1.8.0

Reply:
It's just for the sake of consistency. All other items in the listed deprecation cycles have a full release number
And I don't think it changes the meaning, but it's up to you

Reply:
Then I'd use >= 1.8 in other places for consistency.